Build fails for x86_64-linux-android

https://travis-ci.org/rust-lang-nursery/rustup.rs/jobs/261063687#L1537

```
error[E0308]: mismatched types
  --> target/cargo-home/registry/src/github.com-1ecc6299db9ec823/net2-0.2.30/src/socket.rs:40:51
   |
40 |             ::cvt(c::bind(self.inner.raw(), addr, len as len_t)).map(|_| ())
   |                                                   ^^^^^^^^^^^^ expected i32, found u32
```

I assume the fix is just to 

```diff
-#[cfg(all(target_arch = "aarch64",target_os = "android"))]
+#[cfg(all(any(target_arch = "aarch64", target_arch = "x86_64"),target_os = "android"))]
use libc::c_int as len_t;
```
I'll see if that works locally.

--- a/src/socket.rs
+++ b/src/socket.rs
@@ -30,9 +30,9 @@ impl Socket {
     }
 
     pub fn bind(&self, addr: &SocketAddr) -> io::Result<()> {
-        #[cfg(not(all(target_arch = "aarch64",target_os = "android")))]
+        #[cfg(not(all(any(target_arch = "aarch64", target_arch = "x86_64"), target_os = "android")))]
         use sys::c::socklen_t as len_t;
-        #[cfg(all(target_arch = "aarch64",target_os = "android"))]
+        #[cfg(all(any(target_arch = "aarch64", target_arch = "x86_64"), target_os = "android"))]
         use libc::c_int as len_t;
 
         let (addr, len) = addr2raw(addr);

